gradio/cli/commands/components/publish.py

```diff
@@ -103,9 +65,9 @@ def _publish(
         Optional[Path], Option(help="Path to the demo directory.")
     ] = None,
     source_dir: Annotated[
-        Optional[Path],
+        Path,
```

Comment:
The source_dir is now required so we can find the repo's README but we upload_src is false to preserve the same behavior